tests/guid.test/runit

```diff
@@ -257,6 +257,77 @@ test_guid_autofill() {
 
 }
 
+test_guid_sp() {
+    echo "Testing having guid(guid) as used as key for multiple tables via stored procedure"
+
+    cdb2sql ${CDB2_OPTIONS} $dbnm default 'drop table if exists t1'
+    cdb2sql ${CDB2_OPTIONS} $dbnm default 'drop table if exists t2'
+
+    cdb2sql ${CDB2_OPTIONS} $dbnm default 'create table t1 { 
+    schema 
+    {
+        byte uid[16] null=yes
+        int  i
+    }
+    keys
+    {
+        "PK"=uid
+    }
+    }'
+
+    cdb2sql ${CDB2_OPTIONS} $dbnm default 'create table t2 { 
+    schema 
+    {
+        int other
+        byte uid[16] null=yes
+        cstring  moredata[10]
+    }
+    keys
+    {
+        "PK"=uid
+    }
+    constraints
+    {
+        "PK" -> <"t1" : "PK">
+    }
+    }'
+
+    u=`cdb2sql --tabs ${CDB2_OPTIONS} $dbnm default 'select guid()'`
+
+    cdb2sql ${CDB2_OPTIONS} $dbnm default - << EOF
+begin
+insert into t1 values ($u, 1)
+insert into t2 values (123, $u, '1234')
+commit
+EOF
+
+    assertcnt t1 1
+    assertcnt t2 1
+
+    echo "add stored procedure"
+    cdb2sql ${CDB2_OPTIONS} $dbnm default - << EOF
+create procedure txnins version 'one' {
+local function main(t1_i, t2_other, t2_moredata)
+  local res, rc = db:exec("select guid() as H")
+  local id = res:fetch()       -- id is a table with one column 'H'
+  db:emit(id.H)
+  local t1 = db:table('t1') 
+  local t2 = db:table('t2') 
+  db:begin()
```

Comment:
Explicit transaction is not strictly necessary, is it? Just wondering.. Entire procedure should execute as one transaction. If you do an explicit transaction, you can count the rows right after commit in the procedure.

You might even add all these permutations to this test, although I'm sure it is tested elsewhere.

Reply:
Added db:guid() and extra example without txn.
Thank for suggestion.